superset/models/core.py

```diff
@@ -379,6 +379,9 @@ def _log_query(sql: str) -> None:
 
         with closing(engine.raw_connection()) as conn:
             cursor = conn.cursor()
+            if hasattr(cursor, 'fetch_size'):
```

Comment:
Adding an engine specific configuration on engine agnostic code is not the best path IMO. Also overriding a database connection behind the scenes does not seem like a right way to go

Reply:
I agree that this should be better handled in engine specific code but I could not identify any method where this could be done. I am gladly accepting advise on how to improve this. 

As for the context, I need to use the DeckGL chart to display a large set (up to several hundreds of thousands) 
 of earthquakes along with other key metrics on a dashboard. When removing the limit in the chart config (or when setting the limit to 50000), I am still fetching only 10000 events due to the default fetch_size set in elasticsearch-dbapi.

Thank you for looking into this!

Reply:
I reverted the original change in models/core.py and am now setting fetch_size of the Elastic cursor in the elasticsearch db_engine_spec. Is that the proper implementation?

Reply:
wouldn't it make more sense to just set your database connection to whatever top value you need?
I mean why not just configure the elasticsearch connection with the fetch_size for ROW_LIMIT?

Reply:
According to the elasticsearch-dbapi docs (https://github.com/preset-io/elasticsearch-dbapi#fetch-size), the fetch_size is set at the cursor level and not when initiating a connection. As far as I understand the code (see https://github.com/apache/superset/blob/a330b664c13e34a4c7ca954f9a3a527a43a5535f/superset/models/core.py#L404), there's no method that I could override to create a cursor with engine specific settings.